ext/RMagick/rmdraw.c

```diff
@@ -853,7 +853,15 @@ VALUE Draw_annotate(
     // allowing the app a chance to modify the object's attributes
     if (rb_block_given_p())
     {
-        rb_obj_instance_eval(0, NULL, self);
+        if (rb_proc_arity(rb_block_proc()) == 0)
+        {
+            // Run the block in self's context
```

Comment:
@akicho8 would you mind putting deprecation warnings in these places? Something like:

```c
rb_warning("passing a block without an image argument is deprecated");
```

Reply:
Added warnings was checked from being displayed.
`rspec -w spec/rmagick/image/new_spec.rb`

Reply:
Ah, sorry, I think we actually want `rb_warn`, not `rb_warning`.